### ISSUE TITLE
cfflib - conversion to version 2 throws an exception

### ISSUE THREAD

**Issue by acbellini** [fonttools/fonttools#3293]
In this line:

https://github.com/fonttools/fonttools/blob/98242634c45cf4ca9f8aff6fd059de9dcf734471/Lib/fontTools/cffLib/__init__.py#L450

The code changes the collection it's iterating on, causing an exception.

**Comment by anthrotype** [comment]
thanks, yes that looks wrong. Have you got a test font that exposes the bug?
I believe this occurs only when the font that is being converted to CFF2 has a CID-keyed CFF 1.0 table, i.e. one which has a FDArray attribute

**Comment by acbellini** [comment]
Yes, I hadn't added it because I am not 100% sure I'm doing things right, but here it is with a code snippet. 

The font is extracted from a PDF file, it's a subset of HelveticaNeueLTStd-Thin:

[ff3.cff.zip](https://github.com/fonttools/fonttools/files/12848334/ff3.cff.zip)

And this is the code I'm using:
I don't have a parent TT font really, so I'm creating one and setting its glyphorder in order to proceed. 

```
import fontTools.cffLib as cff
import fontTools.ttLib as tt

with open("ff3.cff", "rb") as file:

    otFont = tt.TTFont()
    cFont = cff.CFFFontSet()
    cFont.decompile(file, otFont)
    font0: cff.TopDict = cFont[cFont.fontNames[0]]
    otFont.setGlyphOrder(font0.getGlyphOrder())
    cff2 = cFont.convertCFFToCFF2(otFont)
```

If incidentally you can point me in the right direction (I need to extract glyphs so that I can draw them with an svgPen) it would be great :D

**Comment by anthrotype** [comment]
> I need to extract glyphs so that I can draw them with an svgPen

in that case, there's no need to convert the CFF to CFF2 first. But the fact that you only have a CFF table without the font containing it makes it a bit tricky to decompile the table properly.. It may be possible to hack your way around that, but I have never tried.

**Comment by acbellini** [comment]
>  But the fact that you only have a CFF table without the font containing it makes it a bit tricky to decompile the table properly.. It may be possible to hack your way around that, but I have never tried.

Yes, I'm realising that. I may end up converting the file from the command line with fontforge, it worked for this file... not sure for others :) Thanks anyway!

**Comment by khaledhosny** [comment]
This seems to work now.
